tests/test_spatial_analysis/test_calculate_spatial_distance.py

```diff
@@ -0,0 +1,200 @@
+            calculate_spatial_distance(
+                adata=self.adata,
+                annotation='cell_type',
+                subset=['nonexistent_image'],
+                imageid='imageid',
+                verbose=False
+            )
+
+    def test_missing_coordinates(self):
+        """Test that the function raises a KeyError
+        when spatial coordinates are missing."""
+        # Remove the spatial coordinates
+        del self.adata.obsm['spatial']
+        with self.assertRaises(KeyError):
+            calculate_spatial_distance(
+                adata=self.adata,
+                annotation='cell_type',
+                verbose=False
+            )
+
+    def test_missing_coordinate_values(self):
+        """Test that the function raises a ValueError
+        when coordinate values are missing."""
+        # Set first cell's x-coordinate to NaN
+        self.adata.obsm['spatial'][0, 0] = np.nan
+        with self.assertRaises(ValueError):
+            calculate_spatial_distance(
+                adata=self.adata,
+                annotation='cell_type',
+                verbose=False
+            )
+
+    def test_invalid_coordinate_dimensions(self):
+        """Test that the function raises a ValueError when coordinates have
+        insufficient dimensions."""
+        # Set spatial coordinates to have only one dimension
+        self.adata.obsm['spatial'] = np.random.rand(100, 1)
+        with self.assertRaises(ValueError):
+            calculate_spatial_distance(
+                adata=self.adata,
+                annotation='cell_type',
+                verbose=False
+            )
+
+    def test_obsm_key_not_found(self):
+        """Test that passing a non-existent obsm_key raises a KeyError."""
+        with self.assertRaises(KeyError):
+            calculate_spatial_distance(
+                adata=self.adata,
+                annotation='cell_type',
+                obsm_key='nonexistent_obsm',
+                verbose=False
+            )
+
+    def test_custom_obsm_key(self):
+        """Test that the function works with a custom obsm_key."""
+        # Rename spatial coordinates
+        self.adata.obsm['custom_spatial'] = self.adata.obsm.pop('spatial')
+        calculate_spatial_distance(
+            adata=self.adata,
+            obsm_key='custom_spatial',
+            annotation='cell_type',
+            verbose=False
+        )
+        self.assertIn('spatial_distance', self.adata.uns)
+
+    def test_z_coordinate_handling(self):
+        """Test that the function correctly handles Z-coordinate data."""
+        # Add a Z-coordinate
+        self.adata.obsm['spatial'] = np.random.rand(100, 3)
+        calculate_spatial_distance(
+            adata=self.adata,
+            annotation='cell_type',
+            verbose=False
+        )
+        result = self.adata.uns['spatial_distance']
+        self.assertIsInstance(result, pd.DataFrame)
+
+    def test_label_parameter(self):
```

Comment:
@fangliu117 , that I/O test can be combined with the first test.

Reply:
yes, combined.